### PR TITLE
Dapp - desktop version

### PR DESCRIPTION
**Pull request by lubej**

### CODE REVIEW

File: src/pages/Wrapper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  value: BigNumber.from(25),
+}, {
+  label: '50%',
+  value: BigNumber.from(50),
+}, {
+  label: 'Max',
+  value: BigNumber.from(100),
+}]
+
+export const Wrapper: FC = () => {
```
Comment (lukaw3d) [inline src/pages/Wrapper/index.tsx:29]: Should we reload the page if user switches network after getting to the wrapper page? https://docs.metamask.io/wallet/reference/provider-api/#chainchanged

File: src/providers/Web3Provider.tsx [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+    const { sapphireEthProvider } = state
+
+    if (!sapphireEthProvider) {
+      throw new Error('[sapphireEthProvider] not initialized!')
+    }
+
+    await sapphireEthProvider.waitForTransaction(txHash)
+
+    return await sapphireEthProvider.getTransaction(txHash)
```
Comment (lukaw3d) [inline src/providers/Web3Provider.tsx]: Doesn't check if transaction failed
```tsx
    const receipt = await sapphireEthProvider.waitForTransaction(txHash)
    if (receipt.status === 0) throw new Error('Transaction failed')
    // or  return transaction + success status
```

File: src/components/Account/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import { useWeb3 } from '../../providers/Web3Provider'
+import { StringUtils } from '../../utils/string.utils'
+
+interface JazzIconProps {
+  address: string;
+  size: number;
+}
+
+const JazzIcon: FC<JazzIconProps> = ({ address, size }) => {
+  const seed = parseInt(address.slice(2, 10), 16)
```
Comment (lukaw3d) [inline src/components/Account/index.tsx]: I would extract this to a function and comment, e.g. https://github.com/oasisprotocol/oasis-wallet-ext/blob/394656d6d0557c473412a8dc15657998bd36367b/src/popup/component/AccountIcon/index.js#L20-L25

File: src/components/Account/index.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  const jazzIconJsx = createJazzIcon(size, seed)
+
+  /* TODO: UNSAFE, consider replacing the library to avoid dangerouslySetInnerHTML */
+  return <div style={{ width: size, height: size }} id={seed.toString()} className={classes.jazzIcon}
+              dangerouslySetInnerHTML={{ __html: jazzIconJsx.outerHTML }} />
```
Comment (lukaw3d) [inline src/components/Account/index.tsx]: At some point we should decide if this is better, and make it consistent
- https://github.com/oasisprotocol/explorer/blob/3a6ef6a2ae2bc19da9e4a09f149771e49bf8722c/src/app/components/JazzIcon/index.tsx#L10-L22
- https://github.com/oasisprotocol/oasis-wallet-web/blob/b2b7374509da94559a1ec3c23cda35c656f0b3f2/src/app/components/JazzIcon/index.tsx#L10-L22

File: src/components/Account/index.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  address: string;
+  networkName: string;
+}
+
+export const Account: FC<Props> = ({ address, networkName }) => {
+  const { state: { explorerBaseUrl } } = useWeb3()
+
+  const handleAccountClick = () => {
+    if (explorerBaseUrl) {
+      window.open(StringUtils.getAccountUrl(explorerBaseUrl, address), '_blank', 'noopener,noreferrer')
```
Comment (lukaw3d) [inline src/components/Account/index.tsx:17]: For after the deadline: these should be links, not onClick listeners

File: src/providers/Web3Provider.tsx [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+  const _addNetwork = async (chainId: number) => {
+    if (chainId === 0x5afe) {
+
+      // Default to Sapphire Mainnet
+      await window.ethereum.request?.({
+        method: 'wallet_addEthereumChain',
+        params: [
+          {
+            chainId: '0x5afe',
+            chainName: 'Sapphire Mainnet',
```
Comment (lukaw3d) [inline src/providers/Web3Provider.tsx]:
```suggestion
            chainName: 'Oasis Sapphire',
```
This doesn't match chainlist data, so metamask shows a warning
(also doesn't match our docs https://github.com/oasisprotocol/docs/blob/c7a21a48ed5d1fcf360e341be2ef4499c071a319/docs/dapp/sapphire/README.mdx#L49-L85 )

File: src/pages/Wrapper/index.tsx (hunk 1 above)
Comment (lubej) [reply src/pages/Wrapper/index.tsx:29]: Yeah, couple of edge cases are not handled:
- network switch;
- account switch;
- user can disconnect from the dapp(Which should take him back to the "Connect Wallet");

Will handle this in separate PR.

File: src/providers/Web3Provider.tsx (hunk 2 above)
Comment (lubej) [reply src/providers/Web3Provider.tsx]: Nice catch 👍

File: src/components/Account/index.tsx (hunk 4 above)
Comment (lubej) [reply src/components/Account/index.tsx]: Yeah, definitely better. Will replace.

File: src/providers/Web3Provider.tsx (hunk 6 above)
Comment (lubej) [reply src/providers/Web3Provider.tsx]: Nice catch 👍

Comment (lubej) [reply src/providers/Web3Provider.tsx]: New explorer URL should probably be fine, right?

Comment (lukaw3d) [reply src/providers/Web3Provider.tsx]: yeah

**Comment by lubej** [comment]
To not make this PR any bigger than it already is, I would merge it as is @lukaw3d

Needs the following(I would rather do it in separate PRs):
- prettier format across all files
- typings are not perfect right now 

But I can fix it in this one as well, your call.

**Review by lukaw3d — APPROVED** [review]

File: src/components/Account/index.tsx (hunk 5 above)
Comment (lubej) [reply src/components/Account/index.tsx:17]: Resolved in https://github.com/oasisprotocol/dapp-wrose/pull/15

**Pull request merged**